File: kotlin/android_java/app/src/main/java/com/example/retrofit_ex/RetrofitInterface.java
package com.example.retrofit_ex;


import java.util.HashMap;

import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.HTTP;
import retrofit2.http.POST;



public interface RetrofitInterface {
    @POST("/login")
    Call<LoginResult> executeLogin(@Body HashMap<String, String> map);

    @POST("/signup")
    Call<Void> executeSignup (@Body HashMap<String, String> map);

    @POST("/post")
    Call<Void> executePost (@Body HashMap<String, Object> map);

    @GET("/post/all")
    Call<ResponseBody> getPost();

    @HTTP(method = "DELETE", path = "/post/delete", hasBody = true)
    Call<Void> deletePost(
            @Body HashMap<String, String> map);

    @POST("/post/update")
    Call<UpdateResult> executeUpdate (@Body HashMap<String, Object> map);

    @POST("/comment/all")
    Call<ResponseBody> getComment(@Body HashMap<String, Object> map);

    @POST("/comment/post")
    Call<Void> executeCommentPost (@Body HashMap<String, Object> map);

    @HTTP(method = "DELETE", path = "/comment/delete", hasBody = true)
    Call<Void> deleteComment(
            @Body HashMap<String, String> map);

    @HTTP(method = "DELETE", path = "/user/delete", hasBody = true)
    Call<Void> deleteUser(
            @Body HashMap<String, String> map);

}

File: kotlin/app/src/main/java/com/example/retrofit_ex/RetrofitInterface.kt
package com.example.retrofit_ex

import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.HTTP
import retrofit2.http.POST
import java.util.*

interface RetrofitInterface {
    @POST("/login")
    fun executeLogin(@Body map: HashMap<String, String>): Call<LoginResult?>?

    @POST("/signup")
    fun executeSignup(@Body map: HashMap<String, String>): Call<Void?>?

    @POST("/post")
    fun executePost(@Body map: HashMap<String?, Any?>?): Call<Void?>?

    @get:GET("/post/all")
    val post: Call<ResponseBody?>?

    @HTTP(method = "DELETE", path = "/post/delete", hasBody = true)
    fun deletePost(
        @Body map: HashMap<String?, String?>?
    ): Call<Void?>?

    @POST("/post/update")
    fun executeUpdate(@Body map: HashMap<String?, Any?>?): Call<UpdateResult?>?

    @POST("/comment/all")
    fun getComment(@Body map: HashMap<String?, Any?>?): Call<ResponseBody?>?

    @POST("/comment/post")
    fun executeCommentPost(@Body map: HashMap<String?, Any?>?): Call<Void?>?

    @HTTP(method = "DELETE", path = "/user/delete", hasBody = true)
    fun deleteUser(
        @Body map: HashMap<String?, String?>?
    ): Call<Void?>?

    @HTTP(method = "DELETE", path = "/comment/delete", hasBody = true)
    fun deleteComment(
        @Body map: HashMap<String, String>
    ): Call<Void?>?
}